client/components/infinite-scroll/index.js

```diff
@@ -20,12 +20,17 @@ export default class InfiniteScroll extends Component {
 	);
 
 	componentDidMount() {
+		this.initialMount = true;
 		window.addEventListener( 'scroll', this.checkScrollPositionHandler );
-		this.checkScrollPosition( false );
+		defer( () => this.checkScrollPosition( false ) );
 	}
 
 	componentDidUpdate() {
-		this.checkScrollPosition( false );
+		if ( this.initialMount ) {
+			this.initialMount = false;
```

Comment:
What is the rationale behind this check? When does it happen that `componentDidMount` is immediately followed by `componentDidUpdate`?

Could the `checkScrollPosition` in `componentDidUpdate`  also be deferred? There can be plenty of `componentDidUpdate`s in rapid succession when components rerender often.

Would it be useful if the used `throttle` or `debounce` to defer the `checkScrollPosition` calls and rate-limit them on frequent updates? That's just an idea.

Reply:
I seem to have misinterpreted the React component lifecycle. There must be a different reason why I was getting a `componentDidUpdate` after the `componentDidMount`, perhaps as a result of one of the other issues you're mentioning.

I'll investigate further, and look into throttling or debouncing `checkScrollPosition` too.